### ISSUE TITLE
Whenever a parameter is optional ot not should be stored in the AST

### ISSUE THREAD

**Issue by Pajn** [eslint/typescript-eslint-parser#186]
e in this example should have an `optional: true` flag
```typescript
function processEntity(e?: Entity) {
    validateEntity(e);
    let s = e!.name;
}
```
